vowpalwabbit/CMakeLists.txt

```diff
@@ -356,6 +356,10 @@ if(BUILD_EXTERNAL_PARSER)
   set(vw_all_sources ${vw_all_sources} ${external_parser_sources})
 endif()
 
+if (vw_BUILD_NETFX)
+set(vw_all_sources ${vw_all_sources} spanning_tree.cc)
```

Comment:
This is needed so that the object is present in the native-only lib, because we cannot import <future> into a /clr-compiled C++ object. There is a [wrapper here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/cs/cli/spanning_tree_clr.h#L16) in the CLI bindings, but if the regular spanning_tree object is missing from libvw linking fails. The Java bindings [pull in spanning_tree.cc directly](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/java/CMakeLists.txt#L40), and it is [pulled in by the spanning_tree binary](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/cluster/CMakeLists.txt#L5) as well.